ctru-rs/src/applets/swkbd.rs

```diff
@@ -577,10 +585,13 @@ impl SoftwareKeyboard {
         self.state.valid_input = ValidInput::FixedLen.into();
     }
 
-    // A reimplementation of `swkbdInputText` from `libctru/source/applets/swkbd.c`. Allows us to
-    // get text from the software keyboard and put it directly into a `String` without requiring
-    // an intermediate fixed-size buffer
-    fn swkbd_input_text(&mut self, output: &mut String) -> SwkbdButton {
+    /// A reimplementation of `swkbdInputText` from `libctru/source/applets/swkbd.c`. Allows us to fix various
+    /// API nits and get rid of awkward type conversions when interacting with the Software Keyboard.
+    ///
+    /// # Safety
+    ///
+    /// The [`Apt`] and [`Gfx`] services must be active when this function is called.
```

Comment:
Is there a way to avoid making this unsafe? Can it take them as params, or maybe in the struct constructor?